File: python源文件/TP2/TP2-2/xianshi.py
import numpy as np
import matplotlib.pyplot as plt
plt.rcParams['font.sans-serif'] = ['SimHei']  # 用来正常显示中文标签
plt.rcParams['axes.unicode_minus'] = False  # 用来正常显示负号

def solve_explicit(rho, Lambda, beta, lambda_, h):
    t = np.arange(0, 1+h, h)
    n = np.zeros(len(t))
    C = np.zeros(len(t))

    # 定义方程组
    def Fun1(t, n, C):
        return (rho - beta) * n / Lambda + lambda_ * C

    def Fun2(t, n, C):
        return beta * n / Lambda - lambda_ * C

    # 定义初值
    n[0] = 1
    C[0] = 1

    # 更新下一个dn/dt和dC/dt
    for i in range(1, len(t)):
        n[i] = n[i-1] + h * Fun1(t[i-1], n[i-1], C[i-1])
        C[i] = C[i-1] + h * Fun2(t[i-1], n[i-1], C[i-1])

    return t, n, C

# 定义参数
rho = 0.0022
Lambda = 10**(-3)
beta = 0.0065
lambda_ = 0.078
h = 0.001

# 调用求解函数
t, n, C = solve_explicit(rho, Lambda, beta, lambda_, h)

# 绘图
plt.figure(figsize=(10, 6))
plt.grid(True)
plt.plot(t, n, 'rh', markersize=3, label='$n(t)$')
plt.plot(t, C, 'bo', markersize=3, label='$C(t)$')
plt.xlabel('t / s')
plt.legend()
plt.title(f'显式法求解中子动力学方程，步长：{h}')
plt.show()

File: matlab源文件/TP 2/TP2-2/xianshi.m
function [n,C]=xianshi(h)
rho=0.0022;Lambda=10^(-3);beta=0.0065;lambda=0.078;%定义参数
t=[0 : h :1];%定义时间范围
n=zeros(size(t));
C=zeros(size(t));
len = length(t);
function n1 =Fun1(t,n,C)%定义方程组
n1=(rho-beta).*n/Lambda+lambda.*C;
end
function C1=Fun2(t,n,C)
C1=beta.*n./Lambda-lambda.*C;
end
%定义初值
n(1)=1;      
C(1)=1;      
for i=2:len 
n(i)=n(i-1)+h*Fun1(t(i-1),n(i-1),C(i-1));% 更新下一个dn/dt
C(i)=C(i-1)+h*Fun2(t(i-1),n(i-1),C(i-1));% 更新下一个dC/dt
end
hold on;
grid on;
plot(t, n,'rh',t,C,'bo',LineWidth=1,MarkerSize=3);		
xlabel('t/s');
legend('$n(t)$','$C(t)$');
title("显式法求解中子动力学方程",Interpreter="none");
fig=gcf;
fig.Children(1).Interpreter='latex';
fig.Children(1).Title.Interpreter='latex';
fig.Children(1).FontSize=13.5;
fig.Children(2).FontSize=15;
end


